File: leku/src/main/java/com/schibsted/leku/LocationPicker.java
package com.schibsted.leku;

import com.schibsted.leku.tracker.LocationPickerTracker;

public class LocationPicker {

  private static final LocationPickerTracker EMPTY_TRACKER = eventName -> {
  };

  private static LocationPickerTracker tracker = EMPTY_TRACKER;

  public static void setTracker(LocationPickerTracker tracker) {
    if (tracker == null) {
      throw new IllegalArgumentException("The LocationPickerTracker instance can't be null.");
    }
    LocationPicker.tracker = tracker;
  }

  public static LocationPickerTracker getTracker() {
    if (tracker == null) {
      tracker = EMPTY_TRACKER;
    }
    return tracker;
  }

  public static void reset() {
    tracker = EMPTY_TRACKER;
  }
}

File: leku/src/main/java/com/adevinta/leku/LocationPicker.kt
package com.adevinta.leku

import com.adevinta.leku.tracker.LocationPickerTracker
import com.adevinta.leku.tracker.TrackEvents

object LocationPicker {

    private val EMPTY_TRACKER = EmptyLocationPickerTracker()

    private var tracker: LocationPickerTracker = EMPTY_TRACKER

    fun setTracker(tracker: LocationPickerTracker?) {
        if (tracker == null) {
            throw IllegalArgumentException("The LocationPickerTracker instance can't be null.")
        }
        LocationPicker.tracker = tracker
    }

    fun getTracker(): LocationPickerTracker {
        return tracker
    }

    fun reset() {
        tracker = EMPTY_TRACKER
    }

    class EmptyLocationPickerTracker : LocationPickerTracker {
        override fun onEventTracked(event: TrackEvents) { }
    }
}
